### ISSUE TITLE
Allow users to record aggregate consumption information for a depot

### ISSUE THREAD

**Issue by jniles** [third-culture-software/bhima#4850]
Separate and apart from the designs proposed in the https://github.com/IMA-WorldHealth/bhima/issues/4848 and https://github.com/IMA-WorldHealth/bhima/issues/4811, we need a way to put in "aggregate" data into BHIMA.  This would be for the "Stock Management in Health Zones" push.

This information would be put in by depot for a chosen period.  Basically, the user would chose a period, choose a depot, and be presented with a table of (consumable) medicines that they can use to put in the consumption during that period as well as the number of days of stock out.

The database schema would look like this
Period ID | Inventory UUID | Lot Number | Consumption | Days of Stock Out
-- | -- | -- | --:| --:
202003 | Quinine 500MG | LA2030 | 700 | 2
202003 | Quinine 500MG | XXT2093 | 100 | --
202003 | Amoxy 250MG | NF33001A | 200 | 3

The interface would look something like this:
![Aggregate Stock Consumption Page](https://user-images.githubusercontent.com/896472/91189841-d5af6980-e6ea-11ea-9527-728659f98942.png)

Basically, the user would put in these two columns and hit "submit" and that would be it.

**How does this fit into the regular design of BHIMA?**

As @jmcameron pointed out, we actually can check this table for consumption information and add it to our regular algorithms (obtained either by https://github.com/IMA-WorldHealth/bhima/issues/4848 orhttps://github.com/IMA-WorldHealth/bhima/issues/4811).  This would make it much easier to design and test.  In a regular hospital setting, this will never be used and will always give 0.

**Comment by jniles** [comment]
The problem with this is that we don't have a mechanism to decrease the quantity in stock by lot.  Since "lot numbers" map to expiration dates, we lose the ability to track expiration.  We may need a "lot number" column in here.  But then, the "days of stock out" would need to be the same for each inventory item...

**Comment by jniles** [comment]
After some discussion with the team today, we believe this might be a better design:

![Aggregate Stock Consumption Page V2](https://user-images.githubusercontent.com/896472/96141322-cbdf0100-0ef8-11eb-834d-fd19370d6b20.png)

**Comment by jmcameron** [comment]
It seems like there needs to be a way for the user to select which inventory items would be shown in the list.  And there needs to be some memory of the last set of inventory items.    Each time this form is used, it can start with previous set of items.  There be an "Add inventory" item to the list it can be expanded.  Also each inventory item needs a "trash can" icon somewhere on the line so it can be removed when no longer needed (or no changes in the period).

My key point here is that we want to make this form as efficient to use as possible--so we do not want it to have thousands of items that the user has to sort through.

**Comment by jniles** [comment]
The idea is that we will need to have the quantity in stock of the lots only for the expiration date tracking.

To calculate the CMM/AMC, we'll need the quantity in stock quantity consumed and stock outs all by inventory item.

**Comment by jniles** [comment]
@jmcameron yes, you are exactly right.  This only makes sense for all instances after the first month.  After that, we should know exactly what is in the health center and can display it here.

The question is - do we control all the medicines in a health center (FOSA), or is there going to be outside ones as well? We need to get that answer clarified from our pharmacy team.

**Comment by jniles** [comment]
@kwilu claim

**Comment by jniles** [comment]
@lomamech is working on this.

@lomamech, I think this should write to `stock_movement`, not `stock_movement_status`.  Otherwise, it will be overridden.

**Comment by lomamech** [comment]
I am in the process of discussing with @mbayopanda , on the different scenarios, concerning the role of this form

**Comment by lomamech** [comment]
- For example how to record the days of breaks in stock_ movement
- What to do with stocks from other suppliers and if these products are not in the list of products in BHIMA

**Comment by jniles** [comment]
For the first, I would just subtract that number of days from the end of the month when you write the movement.

For the second point... I would just ignore them for now.  But I'm curious if you guys can come up with better solutions.

**Comment by lomamech** [comment]
I first think of a form on the client side to transmit the data to the servers then decided on the logic to put in place to record the data in stock movements and stocks movements status

**Comment by jniles** [comment]
Sounds good to me :+1:

**Comment by lomamech** [comment]
We no longer referred to the number of days of stock consumption, is there a reason ??

**Comment by jniles** [comment]
Yeah, the two "correct" algorithms (MSH and algo 3) for calculating the CMM don't use it.  So I don't see a reason to collect it.  We only need to know the days when we have stock, which can be obtained from the days of stock out.
